code/game/mecha/mecha.dm

```diff
@@ -1341,6 +1341,9 @@
 	if(istype(I, /obj/item/pda))
 		var/obj/item/pda/pda = I
 		I = pda.id
+	else if(istype(I, /obj/item/storage/wallet))
+		var/obj/item/storage/wallet/wallet = I
+		I = wallet.front_id
```

Comment:
It would be better to use the GetID helper here
```suggestion
	I = I.GetID()
```

Reply:
I.GetID() causes a runtime error if the person trying to enter doesn't have an ID, any way to resolve this?

Reply:
Only by modifying the GetID() proc i imagine

Reply:
Nevermind I figured it out (I think)

Reply:
Sure, see my most recent review with I?.GetID()